pkg/api/validation/validation.go

```diff
@@ -607,7 +621,7 @@ func validateEnvVarValueFrom(ev api.EnvVar) errs.ValidationErrorList {
 	return allErrs
 }
 
-var validFieldPathExpressions = util.NewStringSet("metadata.name", "metadata.namespace")
+var validFieldPathExpressions = util.NewStringSet("metadata.name", "metadata.namespace", "metadata.labels", "metadata.annotations")
```

Comment:
There needs to be a separate list of valid expressions for env var sources and volumes.  We don't want to expose labels / annotations via env.